aioshelly/coap.py

```diff
@@ -69,6 +86,74 @@ def socket_init():
     return sock
 
 
+def verify_l2socket_creation_permission():
+    """Create a socket using the scapy configured l2socket.
+
+    Try to create the socket
+    to see if we have permissions
+    since sniff will do it another
+    thread so we will not be able to capture
+    any permission or bind errors.
+    """
+    conf.L2socket()
+
+
+class MulticastQuerier:
+    """Multicast querier management."""
+
+    def __init__(self):
+        """Initialize multicast querier thread."""
+        try:
+            verify_l2socket_creation_permission()
+        except (Scapy_Exception, OSError) as ex:
+            if os.geteuid() == 0:
+                _LOGGER.error("Cannot watch for multicast query packets: %s", ex)
+            else:
+                _LOGGER.debug(
+                    "Cannot watch for multicast query packets without root or CAP_NET_RAW: %s",
+                    ex,
+                )
+            return
+        _LOGGER.debug("Multicast querier thread started")
+        self.stop_thread = False
+        self.thread = threading.Thread(target=self.run)
+        self.thread.start()
+
+    def run(self):
+        """Start sniffing for multicast query."""
+        filter_igmp_query = f"igmp and igmp[0] == {int(MULTICAST_QUERY_IGMPTYPE)} and (igmp[4:4] == {int(ipaddress.IPv4Address(MAIN_MULTICAST_IP))} or igmp[4:4] == 0)"
+        pkt_snd = IP(dst=MULTICAST_QUERY_GRP) / IGMP(
+            type=MULTICAST_QUERY_IGMPTYPE,
+            mrcode=100,
+            gaddr=MAIN_MULTICAST_IP,
+        )
+        while not self.stop_thread:
+            result = sniff(
+                filter=filter_igmp_query,
+                store=1,
+                count=1,
+                timeout=MULTICAST_QUERY_TIMEOUT,
```

Comment:
How long is this timeout? Because now Home Assistant shut down will wait for this timeout.

Reply:
Timeout is 4 minutes, but using `self_stop_thread = True` seems to work fine during shutdown and not slow down.
Any better suggestion to close the thread ?

Reply:
Mark it as a daemon and it won't prevent Python from shutting down.

Reply:
Done